pkg/planner/core/logical_limit.go

```diff
@@ -90,6 +90,12 @@ func (p *LogicalLimit) PruneColumns(parentUsedCols []*expression.Column, opt *op
 		return nil, err
 	}
 	p.SetSchema(nil)
+	// If all of its columns are pruned, we need to use its child's output columns.
+	// Make sure that it output at least one column.
+	if len(savedUsedCols) == 0 {
+		p.InlineProjection(p.Children()[0].Schema().Columns, opt)
+		return p, nil
+	}
```

Comment:
<img width="955" alt="image" src="https://github.com/user-attachments/assets/4d89f97f-5e40-42ba-8474-23baa52dd7cc">
L92 set the schema as nil， inlineProjection‘s used array will always be empty？